adapters/python/sources/message_dump_player.py

```diff
@@ -0,0 +1,162 @@
+import signal
+import time
+from typing import Optional, Tuple
+
+import msgpack
+from savant_rs.utils.serialization import (
+    Message,
+    load_message_from_bytes,
+)
+from savant_rs.zmq import BlockingWriter, WriterConfigBuilder
+
+from adapters.shared.thread import BaseThreadWorker
+from savant.utils.config import opt_config, strtobool
+from savant.utils.config import req_config
+from savant.utils.logging import get_logger, init_logging
+from savant.utils.welcome import get_starting_message
+
+LOGGER_NAME = 'adapters.message_dump_player'
+
+logger = get_logger(LOGGER_NAME)
+
+
+class Config:
+    """Configuration for the adapter."""
+
+    def __init__(self):
+        self.playlist_path = req_config('PLAYLIST_PATH')
+        self.zmq_endpoint = req_config('ZMQ_ENDPOINT')
+        self.sync_output = opt_config('SYNC_OUTPUT', False, strtobool)
+
+
+class MessageDumpReader:
+    """Reads messages from the message dump file."""
+
+    def __init__(self, file_path: str):
+        self._list_file = open(file_path, 'rb')
+        self._message_dump_file = None
+        self._unpacker = None
+        self._next_message_dump_file()
+
+    def read(self) -> Optional[Tuple[int, str, Message, bytes]]:
+        """Unpack the next message from the file."""
+
+        if not self._unpacker:
+            return None
+        message = None
+        while message is None:
```

Comment:
Rewrite without loop. Logic is difficult to trace.

Reply:
I cannot handle the case if some files are empty without the loop. Should I throw an exception instead of skipping?